### PR TITLE
Test add all auth

### LINKED ISSUE #2: Add tests for all /auth controllers
There are 4 controllers for `/auth`
- /register, /user, /login and /logout
- Provide a proper test suite for all of these endpoints

### PR DESCRIPTION
**Pull request by navjot-chahal**
### What this PR does:
- Adds mocha tests for `/auth/register`, `/auth/login`, `/auth/user` and `/auth/logout` routes.
- Changes logout request type to "POST" from "GET" for both server and client side.

### Screenshots / Videos:
- CLI mocha test results:
Screen Shot 2021-10-19 at 7 43 45 PM

### Any information needed to test this feature:
- Use `npm run dev` to start the test in the server folder.

closes #2

### CODE REVIEW

File: server/test/auth.test.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    describe("/POST logout", () => {
+      it("logout should clear cookie and text message", async () => {
+        const res = await agent.post(`/auth/logout`);
+        res.text.should.include("You have successfully logged out");
+        agent.should.not.have.cookie("token");
+      });
+    });
+  });
+
+  // Delete the test user from database
```
Comment (ezeilo-su) [inline server/test/auth.test.js:85]: You can remove this comment as the code is clear what is does.

**Review by sundayezeilo — COMMENTED** [review]
Your tests look good! I Like those screenshots :+1:

File: client/src/helpers/APICalls/logout.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { FetchOptions } from '../../interface/FetchOptions';
 
 const logout = async (): Promise<AuthApiData> => {
   const fetchOptions: FetchOptions = {
-    method: 'GET',
+    method: 'POST',
```
Comment (ugur-akin) [inline client/src/helpers/APICalls/logout.ts:6]: 🙌🏼

File: server/test/auth.test.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        .end((err, res) => {
+          res.should.have.status(201);
+          res.should.have.cookie("token");
+          res.body.should.have.property("success");
+          res.body.success.should.have.property("user");
+          pass();
+        });
+    });
+
+    it("register should return 400 code when user already exists", (pass) => {
```
Comment (ugur-akin) [inline server/test/auth.test.js:34]: `400` is probably not the most appropriate status code for this case (you can find some suggestions on what might be on this [thread](https://stackoverflow.com/questions/9269040/which-http-response-code-for-this-email-is-already-registered/9270432)). 

Ideally, we don't want to define success/failure of our test cases based on the implementation. Therefore I would recommend not looking at the implementation(s) of the feature(s) you're testing as much as possible while designing them (as if to pretend they aren't implemented yet) 👍🏼

File: server/test/auth.test.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+          pass();
+        });
+      });
+    });
+
+    describe("/POST logout", () => {
+      it("logout should clear cookie and text message", async () => {
+        const res = await agent.post(`/auth/logout`);
+        res.text.should.include("You have successfully logged out");
+        agent.should.not.have.cookie("token");
```
Comment (ugur-akin) [inline server/test/auth.test.js:80]: ❤️

**Review by ugur-akin — CHANGES_REQUESTED** [review]
Looks great, thanks for you work 👍🏼 I left a  minor comment, take a look and let's check back and merge this soon! 🎉

**Comment by rajivtitus** [comment]
@navjot-chahal why has this branch been merged without approval? :)

**Comment by navjot-chahal** [comment]
@rajivtitus There was a very minor change. Just the error code response is changed to `409` from `400`. Is there any issue with the last commit?

Also, if this merge was a violation then I will make sure to wait for approval for a minor change as well before merging.

**Comment by rajivtitus** [comment]
@navjot-chahal There are no issues with the last commit, good work here :)

In the future please wait for approval from a Hatchways team member prior to merging PRs. We review PRs once again to make sure the requested feedback has been implemented or any new issues have been introduced.

**Pull request merged**